File: smartexpense-server/src/main/java/com/smartexpense/server/model/SmsTemplate.java
package com.smartexpense.server.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "sms_templates")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SmsTemplate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "sender_pattern", nullable = false)
    private String senderPattern;

    @Column(name = "amount_regex", nullable = false)
    private String amountRegex;

    @Column(nullable = false)
    private String type;

    @Column(name = "bank_name", nullable = false)
    private String bankName;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive;

    @Column(nullable = false)
    private Integer version;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (isActive == null) isActive = true;
        if (version == null) version = 1;
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: app/src/main/java/com/arijit/budgettracker/db/SmsTemplate.kt
package com.arijit.budgettracker.db

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "sms_templates")
data class SmsTemplate(
    @PrimaryKey val id: Long,
    val senderPattern: String,
    val amountRegex: String,
    val type: String,
    val bankName: String,
    val isActive: Boolean = true,
    val version: Int = 1
)
